allowed-breaking-changes.txt

```diff
@@ -1 +1,4 @@
-
+# Actually adding any artifact type will break the load() type signature  because I could have written
```

Comment:
@eladb @rix0rrr I believe Metallica said it best: *Sad but true*.

I guess it makes sense not to use union types when trying to create type-safe schemas, since they are essentially a way to represent **dynamic** (be it finite) properties.

So instead of doing this:

```json
{
  "artifacts": [
    {
      "type": "asset",
      "properties": {
        "asset-prop-1" : "value"
      }
    },
    {
      "type": "stack",
      "properties": {
        "stack-prop-1" : "value"
      }
    }
  ]
}
```

We should be doing something like:

```json
{
  "artifacts": {
    "assets": [{
      "asset-prop-1": "value"
    }],
    "stacks": [
      {
        "stack-prop-1": "value"
      }
    ]
  }
}
```

This way the introduction of a new artifact type results in the addition of a new key, which is a non breaking change.

I vote we do this soon and am happy to take it on, thoughts?

Reply:
Yes let’s do it! I also think this is a better schema.